docs/DeveloperGuide.md

```diff
@@ -4,22 +4,194 @@
+instance. Its priority level is set to the specified priority.
+
+The following sequence diagram shows the relevant interactions behind `setPriorityLevel`:
+
+![setPriorityLevelSequenceDiagram](images/setPriorityLevelSequenceDiagram.png)
+
+#### Study Plan Feature
+
+##### Proposed Implementation
+
+The Study Plan feature is facilitated by `StudyPlanList`.
+
+Internally, it implements a `TreeMap`, with date as key and the list of flashcard indexes to study as value. 
+Key-value pairs in the `TreeMap` are sorted by dates.
+
+It implements the following operations:
+
+- `StudyPlanList#updateStudyPlan()` - Updates the study plan list.
+- `StudyPlanList#getStudyPlanList()` - Gets the list of study plans.
+
+Given below is an example usage scenario and how the study plan mechanism behaves at each step.
+
+Step 1. The user launches the application and an empty `StudyPlanList` is initialized.
+
+Step 2. The user executes `plan` command and the `updateStudyPlan` operation is invoked. The application prompts the
+user for date and the corresponding list of flashcards indexes.
+
+The following sequence diagram shows how the `updateStudyPlan` operation works:
+
+![updateStudyPlanSequenceDiagram](images/updateStudyPlanSequenceDiagram.png)
+
+Step 3. The user executes `show-plan` command and the `getStudyPlanList` operation is invoked. The application
+displays the user's study plan list.
+
+The following sequence diagram shows how the `getStudyPlanList` operation works:
+
+![getStudyPlanListSequenceDiagram](images/getStudyPlanListSequenceDiagram.png)
+
+#### Random Review Feature
+
+##### Proposed Implementation
+
+The random review mechanism is facilitated by `FlashcardList`.It stores an `List<Flashcard>` internally which contains
+all the flashcards created by the user.
+
+It implements the following operation:
+
+- `FlashcardList#reviewRandomFlashcards()` -Randomize all the flashcards, then display the details of them one by one 
+to the user. And update the review status of the flashcards corresponding to the user's' responses.
+
+Given below is an example usage scenario and how the random review mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The `FlashcardsList` will be initialized with the empty 
+flashcards list.
+
+Step 2. The user executes `event/person/other commands` to add multiple flashcards to the flashcard list 
+(The detailed implementation of creating flashcards explained in the Flashcard Creation Feature)
+
+Step 3. The user executes `random command` and `reviewRandomFlashcards()` operation is invoked. The application prompts 
+responses for each unreviewed flashcard that whether the user want to mark it as reviewed or not.
+
+The following sequence diagram shows how the   `reviewRandomFlashcards()` operation works:
+
+![randomReviewSequenceDiagram](images/randomReviewSequenceDiagram.png)
```

Comment:
This sequence diagram is a little bit complicated. Can you simplify it by omitting some less important details?